pkg/network/setup/netstat.go

```diff
@@ -112,13 +112,29 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 	// Guest Agent information will add and conditionally override data gathered from the cache.
 	interfacesStatus = ifacesStatusFromGuestAgent(interfacesStatus, domain.Status.Interfaces)
 
-	primaryInterfaceStatus, interfacesStatus := netvmispec.PopInterfaceByNetwork(interfacesStatus, netvmispec.LookupPodNetwork(vmi.Spec.Networks))
-	if primaryInterfaceStatus != nil {
-		interfacesStatus = append([]v1.VirtualMachineInstanceNetworkInterface{*primaryInterfaceStatus}, interfacesStatus...)
-	}
-
 	interfacesStatus = ifacesStatusFromMultus(interfacesStatus, multusStatusNetworksByName, vmiInterfacesSpecByName)
 
+	if vmiSpecPrimaryNet := netvmispec.LookUpDefaultNetwork(vmi.Spec.Networks); vmiSpecPrimaryNet != nil {
```

Comment:
IIUC the logic below, unlike before, given that the primary network is defined, you always add it to the status, even if it was not detected before (by the domain) or by the vmi controller.

First, I do not understand how this can even occur, but even if it may occur, I do not think we should "lie" about it. We should report only what we detect and preserve specific info (e.g. `podInterfaceName`) from the vmi controller.

I would have expected something like this:
```go
primaryInterfaceStatus, interfacesStatus := netvmispec.PopInterfaceByNetwork(interfacesStatus, netvmispec.LookupPodNetwork(vmi.Spec.Networks))
if primaryInterfaceStatus == nil && prevPrimaryIfaceStatus != nil && vmiSpecPrimaryNet != nil {
        primaryInterfaceStatus = new(v1.VirtualMachineInstanceNetworkInterface)
}
if primaryInterfaceStatus != nil { 
    primaryInterfaceStatus.PodInterfaceName = prevPrimaryIfaceStatus.PodInterfaceName
    interfacesStatus = append([]v1.VirtualMachineInstanceNetworkInterface{*primaryInterfaceStatus}, interfacesStatus...)
}
```

Reply:
The logic was fixed.